library/gui/key_bindings/src/main/java/org/quiltmc/qsl/key/bindings/impl/KeyBindingRegistryImpl.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.key.bindings.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+import net.minecraft.client.option.GameOptions;
+import net.minecraft.client.option.KeyBinding;
+
+import org.quiltmc.qsl.key.bindings.mixin.client.KeyBindingAccessor;
+
+@Environment(EnvType.CLIENT)
+public class KeyBindingRegistryImpl {
+	public static final Logger LOGGER = LogManager.getLogger();
+
+	private static Map<KeyBinding, Boolean> quiltKeys = new HashMap<>();
+	private static List<KeyBindingManager> keyBindingManagers = new ArrayList<>(1);
+	private static KeyBinding[] enabledQuiltKeysArray = new KeyBinding[] {};
+	private static List<KeyBinding> disabledQuiltKeys = new ArrayList<>(0);
+
+	public static KeyBinding registerKeyBinding(KeyBinding key, boolean enabled) {
+		if (key == null) {
+			LOGGER.error("Attempted to register a null key bind!");
+			return null;
+		}
+
+		for (KeyBinding otherKey : quiltKeys.keySet()) {
+			if (key == otherKey || key.getTranslationKey() == otherKey.getTranslationKey()) {
+				LOGGER.error("Attempted to register %s, but a conflicting key has already been registered!", key.getTranslationKey());
```

Comment:
Clarify what `conflicting` means in this context. I would assume the same keybind, but its the translation key.

Reply:
In addition, i would have a separate error for when the same keybinding is registered twice.